shell/plugins/dashboard-store/mutations.js

```diff
@@ -46,15 +46,24 @@ function registerType(state, type) {
 }
 
 export function replace(existing, data) {
+  const existingPropertyMap = {};
+
   for ( const k of Object.keys(existing) ) {
     delete existing[k];
+    existingPropertyMap[k] = true;
   }
 
+  let newProperty = false;
+
   for ( const k of Object.keys(data) ) {
+    if (!newProperty && !existingPropertyMap[k]) {
+      newProperty = true;
+    }
+
     existing[k] = data[k];
   }
 
-  return existing;
+  return newProperty ? reactive(existing) : existing;
```

Comment:
If you do look at the implementation vue does just return an object if it's already a proxy

https://github.com/vuejs/core/blob/main/packages/reactivity/src/reactive.ts#L92-L97

https://github.com/vuejs/core/blob/main/packages/reactivity/src/reactive.ts#L274-L275

Reply:
From my understanding of the bug (which isn't the best) it's new properties on an existing reactive object that fail to notify and that by calling `reactive` again fixed it. Now though i'm not sure. 

Shouldn't calling reactive again then be a no op? Will investigate tomorrow

Reply:
Yep, I think that @richard-cox described the issue well... In Elemental (where I've noticed this) there is a property called `downloadURL` that doesn't exist when the resource get created. Only after a minute or so, we receive the update via websockets and the new property appears, which we need to monitor for reactivity and it's not triggering it.

Reply:
Dug into it some more 

- Couldn't debug step into `reactive`
- Removing the `replace` fn change breaks the solution, adding it back in makes it work again
  - using steps from https://github.com/rancher/dashboard/issues/12189

Given above i think should get this one in and return if there's more reactive issues.

@codyrancher let me know if there's any objections otherwise will merge on Monday

Reply:
I was mostly mentioning it because I thought you could always wrap the object with reactive() and didn't have to check for a newProperty first. 

But I'm still happy for you to ship.